lighthouse-core/gather/driver.js

```diff
@@ -893,8 +894,8 @@ class Driver {
     // These can 'race' and that's OK.
     // We don't want to wait for Page.navigate's resolution, as it can now
     // happen _after_ onload: https://crbug.com/768961
-    this.sendCommand('Page.enable');
-    this.sendCommand('Emulation.setScriptExecutionDisabled', {value: disableJS});
+    await this.sendCommand('Page.enable');
```

Comment:
Is it okay to await these?  The comments above them talk about racing like this might be purposefully not-awaited.

Reply:
Bug is marked as fixed now. If it truly is, can you also remove the comments?

Reply:
the big bug was definitely about not awaiting `Page.navigate`, but why are we needing to `await` these? is it just about unhandled rejections?

Reply:
right I'm onboard with that, but I'm asking because there are then other things we might try instead of serialling awaiting like `this._sendAndSilentlyIgnoreErrors` or something :)

It looks like we used to be just fine waiting on these in sequence though, and the `Page.navigate` change just moved all of 'em to be together, so we don't need to go crazy avoiding the `await` https://github.com/GoogleChrome/lighthouse/pull/3413

I'm a tad spooked by this given that it was a real nasty bug to hunt down, but I think this makes sense if the bug really is fixed. +1 to removing the comments if we're making this change

Reply:
Hmmm, I think I might have explained this wrong?  Or mis-titled this PR.  I want these functions to loudly throw their underlying error (PROTOCOL_TIMEOUT) upwards when they reject their promises.  So I don't want to silently ignore errors I just want to handle the unhandled rejections that `sendCommand` can throw now instead of crashing LH for not handling a rejected-promise. 

Is await the best way to do that?  Or are you suggesting some kind of promise catching alternative?

Reply:
Oh, so we *do* want these two errors to be fatal? Yeah `await`'ing serially like we were before SGTM, sounds like we also want an empty catch on the `Page.navigate` then as well?

Reply:
AFAIK since `Page.navigate` uses `_innerSendCommand` directly instead of the `driver.sendCommand` wrapper which has the timeout configured around it I don't think it can reject?  I was only force rejecting promises from `driver.sendCommand` though in my testing, so...?

interrogative: Do we need protocol timeouts for `_innerSendCommand`? What if those timeout, what happens?  Or is that specifically only called directly when we don't want a timeout e.g. `Page.navigate`?

Reply:
`_innerSendCommand` is specifically for when you don't want to timeout.

I'm not sure if `_innerSendCommand` can ever reject. You could try setting a breakpoint just before, then kill Chrome, then hit play, and see what happens ...

Reply:
Or I guess `sendCommand` is never the real problem it's always the PROTOCOL_TIMEOUT that's the real problem?

Reply:
maybe I'm not following the conversation correctly here, but `_innerSendCommand` can reject via `connection`:

https://github.com/GoogleChrome/lighthouse/blob/9ebf0692de37bd772c59280dd63071bb9357989e/lighthouse-core/gather/connections/connection.js#L120-L124

(throwing inside `Promise.resolve().then(_ => {/* throw */})` rejects it, then `callback.resolve()` with a rejected promise rejects as well. We could probably make that a lot more understandable with an explicit `callback.resolve` *or* `callback.reject`, as well as using an `async` function for the inner part)

Reply:
so do we want to add an empty `catch` to the `this._innerSendCommand('Page.navigate', {url});` line? Or are we assuming it's so rare that it's ok that a `unhandledRejection` listener can catch it?

Reply:
Since it seems like this _never_ happens, should we just let sleeping dragons lie?